cassandra/cassandra-docker.sh

```diff
@@ -250,39 +281,88 @@ else
+    if [ $? != 0 ]; then
+        echo "Failed to create a Java Keystore file with the service-specific certificate. Aborting."
+        exit 1
+    fi
 fi
+sed -i 's#${KEYSTORE_PASSWORD}#'$KEYSTORE_PASSWORD'#g' /opt/apache-cassandra/conf/cassandra.yaml
+sed -i 's#${KEYSTORE_FILE}#'$KEYSTORE_FILE'#g' /opt/apache-cassandra/conf/cassandra.yaml
 
-# handle setting up the truststore
-if [ -n "$TRUSTSTORE_FILE" ]; then
-   sed -i 's#${TRUSTSTORE_FILE}#'$TRUSTSTORE_FILE'#g' /opt/apache-cassandra/conf/cassandra.yaml
+# handle setting up the trust store for the inter node communication
+if [ -n "$TRUSTSTORE_NODES_PASSWORD_FILE" ]; then
+   TRUSTSTORE_NODES_PASSWORD=$(cat $TRUSTSTORE_NODES_PASSWORD_FILE)
 fi
-if [ -n "$TRUSTSTORE_PASSWORD_FILE" ]; then
-   TRUSTSTORE_PASSWORD=$(cat $TRUSTSTORE_PASSWORD_FILE)
+if [ ! -n "$TRUSTSTORE_NODES_FILE" ]; then
+    TRUSTSTORE_NODES_FILE=${TRUSTSTORE_NODES_FILE:-"${KEYSTORE_DIR}/.nodes.truststore"}
+    TRUSTSTORE_NODES_PASSWORD=${TRUSTSTORE_NODES_PASSWORD:-$(openssl rand -base64 512 | tr -dc A-Z-a-z-0-9 | head -c 17)}
+
+    echo "Building the trust store for inter node communication"
+    ${KEYTOOL_COMMAND} -noprompt -import -alias ${SERVICE_ALIAS} -file ${SERVICE_CERT} -keystore ${TRUSTSTORE_NODES_FILE} -trustcacerts -storepass ${TRUSTSTORE_NODES_PASSWORD}
+    if [ $? != 0 ]; then
+        echo "Failed to build the trust store by importing the service CA. Aborting."
+        exit 1
+    fi
 fi
-if [ -n "$TRUSTSTORE_PASSWORD" ]; then
-   sed -i 's#${TRUSTSTORE_PASSWORD}#'$TRUSTSTORE_PASSWORD'#g' /opt/apache-cassandra/conf/cassandra.yaml
+sed -i 's#${TRUSTSTORE_NODES_FILE}#'$TRUSTSTORE_NODES_FILE'#g' /opt/apache-cassandra/conf/cassandra.yaml
+sed -i 's#${TRUSTSTORE_NODES_PASSWORD}#'$TRUSTSTORE_NODES_PASSWORD'#g' /opt/apache-cassandra/conf/cassandra.yaml
+
+# handle setting up the trust store for the client communication
+if [ -n "$TRUSTSTORE_CLIENT_PASSWORD_FILE" ]; then
+   TRUSTSTORE_CLIENT_PASSWORD=$(cat $TRUSTSTORE_CLIENT_PASSWORD_FILE)
 fi
+if [ ! -n "$TRUSTSTORE_CLIENT_FILE" ]; then
+    TRUSTSTORE_CLIENT_FILE=${TRUSTSTORE_CLIENT_FILE:-"${KEYSTORE_DIR}/.clients.truststore"}
+    TRUSTSTORE_CLIENT_PASSWORD=${TRUSTSTORE_CLIENT_PASSWORD:-$(openssl rand -base64 512 | tr -dc A-Z-a-z-0-9 | head -c 17)}
+
+    # the next few lines deserve an explanation: the service-ca.crt provided by OpenShift contains the root CA and the
+    # service-ca certificates in a single file. Java's keytool can't handle this, it seems, and ends up importing only
+    # the first one. So, we split the file, having one cert per resulting file. The next lines are for that, and
+    # will only work properly on the scenario described. If the scenario ever changes, the next lines will probably
+    # need to be adapted accordingly. The best solution would be to have one cert per file.
```

Comment:
> Customers can have their own service-ca.crt values that come from a cert chain, so it is possible that the scenario you describe is invalid. @deads2k or @liggitt may know a way to correctly distinguish between the two cases.

"two cases"?  It's a CA bundle, so it may have an unknown number of certificates inside of it.  Regardless of who makes it (customer or us), I think you end up in the same place.  Take one file and split it into 1..n new files. right?